### PR TITLE
NDEV-712: Implement data flow from Kafka to ClickHouse

### PR DESCRIPTION
**Pull request by Deniskore**

### CODE REVIEW

**Review by ivandzen — CHANGES_REQUESTED** [review]

File: v2-advanced/clickhouse/create_notify_transaction.sql [hunk 1]
```diff
@@ -0,0 +1,31 @@
+CREATE DATABASE IF NOT EXISTS events ON CLUSTER 'events';
+
+CREATE TABLE IF NOT EXISTS events.notify_transaction_local ON CLUSTER '{cluster}' (
+    slot UInt64 CODEC(DoubleDelta, ZSTD),
+    signature Array(UInt8) CODEC(ZSTD),
+    notify_transaction_json String CODEC(ZSTD(5)),
+    timestamp DateTime CODEC(T64, ZSTD(5))
```
[the remark this answers is not in the document]
Comment (Deniskore) [reply v2-advanced/clickhouse/create_notify_transaction.sql]: Done.

File: v2-advanced/clickhouse/create_notify_transaction.sql [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    kafka_topic_list = 'notify_transaction',
+    kafka_group_name = 'clickhouse',
+    kafka_num_consumers = 1,
+    kafka_format = 'JSONAsString';
+
+CREATE MATERIALIZED VIEW IF NOT EXISTS events.notify_transaction_queue_mv ON CLUSTER '{cluster}' TO events.notify_transaction_local AS
+SELECT JSONExtract(notify_transaction_json, 'slot', 'UInt64') AS slot,
+       JSONExtract(notify_transaction_json, 'signature', 'Array(Nullable(UInt8))') AS signature,
+       notify_transaction_json,
+       now() as timestamp
```
[the remark this answers is not in the document]
Comment (Deniskore) [reply v2-advanced/clickhouse/create_notify_transaction.sql]: Done.

File: v2-advanced/clickhouse/create_notify_block.sql [hunk 3]
```diff
@@ -0,0 +1,34 @@
+CREATE DATABASE IF NOT EXISTS events ON CLUSTER 'events';
+
+CREATE TABLE IF NOT EXISTS events.notify_block_local ON CLUSTER '{cluster}' (
+    slot UInt64 CODEC(DoubleDelta, ZSTD),
+    hash String CODEC(ZSTD(5)),
+    notify_block_json String CODEC(ZSTD(5)),
+    timestamp DateTime CODEC(DoubleDelta, ZSTD(5)),
```
[the remark this answers is not in the document]
Comment (Deniskore) [reply v2-advanced/clickhouse/create_notify_block.sql]: Done.

File: v2-advanced/clickhouse/create_notify_block.sql [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    kafka_topic_list = 'notify_block',
+    kafka_group_name = 'clickhouse',
+    kafka_num_consumers = 1,
+    kafka_format = 'JSONAsString';
+
+CREATE MATERIALIZED VIEW IF NOT EXISTS events.notify_block_queue_mv ON CLUSTER '{cluster}' TO events.notify_block_local AS
+SELECT JSONExtract(notify_block_json, 'slot', 'UInt64') AS slot,
+       _key as hash,
+       notify_block_json,
+       now() as timestamp
```
[the remark this answers is not in the document]
Comment (Deniskore) [reply v2-advanced/clickhouse/create_notify_block.sql]: Done.

File: v2-advanced/clickhouse/create_update_slot.sql [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+kafka_group_name = 'clickhouse',
+kafka_num_consumers = 1,
+kafka_format = 'JSONEachRow';
+
+-- ENGINE Should be ReplicatedSummingMergeTree?
+CREATE MATERIALIZED VIEW IF NOT EXISTS events.update_slot_queue_mv ON CLUSTER '{cluster}' to events.update_slot_local AS
+SELECT slot,
+    parent,
+    slot_status,
+    now() as timestamp
```
[the remark this answers is not in the document]
Comment (Deniskore) [reply v2-advanced/clickhouse/create_update_slot.sql:31]: We should postpone this as it requires changes to the plugin, I will do a separate PR for the plugin and for the filter later.

**Review by ivandzen — APPROVED** [review]

**Pull request merged**